pkg/client/events/replay_client.go

```diff
@@ -191,7 +219,20 @@ func (rClient *replayClient[T, R]) retryPublishEventsFactory(ctx context.Context
 			rClient.newMapEventsBytesToTFn(errCh),
 		)
 
-		// Initially set latestObsvbls and update if after retrying on error.
+		// Subscribe to the eventBzObs and block until the channel closes.
+		// Then pass this as an error to force the retry.OnError to resubscribe.
+		go func() {
+			eventsBzSub := eventsBz.Subscribe(ctx)
+			for range eventsBzSub.Ch() {
```

Comment:
Did you forget to push some changes, I thought we did this already?

```suggestion
			eventsBzObserver := eventsBz.Subscribe(ctx)
			for range eventsBzObserver.Ch() {
```